File: couch/configuration/shows/configuration.js
function(doc, req) {
  log(req);

  var name = encodeURIComponent(req.form.key_value);
  var url

  if(name === "sofia.conf"){
    var host = req.form.hostname
    var url = "/fxc/_design/configuration/_list/sofia/sofia_profiles" +
      "?startkey=" + encodeURIComponent('["' + host + '",0]') +
      "&endkey="   + encodeURIComponent('["' + host + '",1]');
  } else {
    var url = "/fxc/_design/configuration/_list/" + name + "/conf";
  }

  return {code: 302, body: '', headers: {'Location': url}}
}

File: model/configuration.rb
module FXC
  class Configuration
    include Makura::Model

    property :name
    property :server

    def self.render(name, hostname, opts = {})
      if name == "sofia.conf"
        Configuration.render_sofia(hostname)
      else
        Configuration.render_common(name, hostname)
      end
    end

    def self.render_common(name, hostname, opts = {})
      keys = [[name, hostname], [name, nil]]
      opts.merge!(:payload => {'keys' => keys}, 'Content-Type' => 'application/json', :limit => 1)
      database.post("_design/configuration/_list/#{name}/conf", opts)
    end

    def self.render_sofia(hostname, opts = {})
      database.get("_design/configuration/_list/sofia/sofia_profiles",
                   startkey: [hostname, 0],
                   endkey:   [hostname, 1])
    end
  end
end
